src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java

```diff
@@ -92,6 +91,66 @@ static LoanPolicy from(JsonObject representation) {
     }
   }
 
+  public HttpResult<Loan> overrideRenewal(Loan loan, DateTime systemDate,
+                                          DateTime overrideDueDate, String comment) {
+    try {
+      if (isNotRenewable()) {
+        return overrideRenewalForDueDate(loan, overrideDueDate, comment);
+      }
+      final HttpResult<DateTime> proposedDueDateResult =
+        determineStrategy(true, systemDate).calculateDueDate(loan);
+
+      final JsonObject loansPolicy = getLoansPolicy();
+      if (proposedDueDateResult.failed() && isFixed(loansPolicy)) {
+        return overrideRenewalForDueDate(loan, overrideDueDate, comment);
+      }
+      if (proposedDueDateResult.failed() && isRolling(loansPolicy)) {
+        DueDateStrategy dueDateStrategy = getRollingRenewalOverrideDueDateStrategy(systemDate);
+        return dueDateStrategy.calculateDueDate(loan)
+          .map(dueDate -> loan.overrideRenewal(dueDate, getId(), comment));
+      }
+      if (proposedDueDateResult.succeeded() &&
+        reachedNumberOfRenewalsLimit(loan) && !unlimitedRenewals()) {
+        return proposedDueDateResult.map(dueDate -> loan.overrideRenewal(dueDate, getId(), comment));
+      }
+      return HttpResult.failed(new ValidationErrorFailure(errorForNotMatchingOverrideCases()));
```

Comment:
My interpretation of this logic is as follows:
Override when the due date calculation failed (for any reason) and the policy is fixed
Override when the due date calculation failed (for any reason) and the policy is rolling
Override when the due date calculation succeeded and the limit has been reached
Otherwise fail the renewal as not for any of the expected reasons

Is that a sensible interpretation?

If so, it would appear that any failure when the policy is either fixed or rolling will allow an override, including if the policy is invalid for some reason. Does that match your understanding?

Reply:
Actually, there is one more case: item is non-renewable.
A few comments to these cases:
If calculation failed (for any reason) and policy is rolling then due date is calculated using special due date strategy (RollingRenewalOverrideDueDateStrategy) that inherits RollingRenewalDueDateStrategy and just skips due date truncation so that avoiding "outside of the range" failure (other failure reasons remain)
Other case is fixed loan policy. The goal here is the same: override only when  "outside of the range" failure occurs. I'm not really sure what is the best way to determine calculation failure reason and how to take a decision based on this reason. But in case of fixed renewal FixedScheduleRenewalDueDateStrategy "outside of the range" is the only reason that may occur. 
And I decided to use it for now, but it can cause problems in the future if due date strategy implementation changes.

Reply:
Agreed, it works for now due to the way the internals of the calculations work. Thank you for helping me think this through.

I think I was mostly looking for a way to raise awareness of this for future developers in this area. It is difficult to recreate in tests due to the coupling of code in this area (which predates this change and likely needs improving).

The alternative is a comment which could likely get out of date, so I won't add that.